### PR TITLE
Closes  #108

### LINKED ISSUE #108: HDF5 initialization speeding up
--------
Issue Description
--------
The initialization of the HDF5 (in case it is in read mode) should be accelerated a bit, for those databases that are quite big. This is required because ```visiting``` the groups can be slow and if the number of groups is high, it can take minutes

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [X] 1. Is it tagged with a type: defect or improvement? ***improvement***
- [X] 2. Is it tagged with a priority: critical, normal or minor? ***normal***
- [X] 3. If it will impact requirements or requirements tests, is it tagged with requirements? ***n/a***
- [X] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users. ***improvement***
- [X] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.) ***yes***

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [ ] 1. If the issue is a defect, is the defect fixed?
- [ ] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [ ] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [ ] 4. If the issue is a defect, does it impact the latest stable branch? If yes, is there any issue tagged with stable (create if needed)?
- [ ] 5. If the issue is being closed without a merge request, has an explanation of why it is being closed been provided?

### PR DESCRIPTION
**Pull request by alfoa**
--------
Pull Requeset Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)
#108 

##### What are the significant changes in functionality due to this change request?
This PR is just a simple optimization of the ```__isGroup``` ***method in h5py_interface_creator.py***. This accelerates the initialization of the HDF5 of a factor of 3.

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [ ] 1. Review all computer code.
- [ ] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [ ] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [ ] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [ ] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [ ] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [ ] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [ ] 8. If an analytic test is changed/added is the the analytic documentation updated/added?

### CODE REVIEW

File: framework/h5py_interface_creator.py [hunk 1]
```diff
@@ -139,12 +140,10 @@ def __isGroup(self,name,obj):
     """
     if isinstance(obj,h5.Group):
       self.allGroupPaths.append(name)
-      self.raiseAMessage('Accessing group named ' +name)
-      if "EndGroup" in obj.attrs:
-        self.allGroupEnds[name]  = obj.attrs["EndGroup"]
-      else:
-        self.raiseAWarning('not found attribute EndGroup in group ' + name + '.Set True.')
+      try   : self.allGroupEnds[name]  = obj.attrs["EndGroup"]
```
Comment (joshua-cogliati-inl) [inline framework/h5py_interface_creator.py]: This should be on two lines.

File: framework/h5py_interface_creator.py [hunk 2]
```diff
@@ -139,12 +140,10 @@ def __isGroup(self,name,obj):
...
+      except:
```
Comment (joshua-cogliati-inl) [inline framework/h5py_interface_creator.py]: Bare excepts are not allowed by the code standard.  This should catch a specifc exception or should get the execption and report it.

Comment (alfoa) [inline framework/h5py_interface_creator.py]: You are perfectly right. 

I address it.

**Comment by alfoa** [comment]
I addressed your comments

File: framework/h5py_interface_creator.py [hunk 3]
```diff
@@ -140,8 +139,9 @@ def __isGroup(self,name,obj):
     """
     if isinstance(obj,h5.Group):
       self.allGroupPaths.append(name)
-      try   : self.allGroupEnds[name]  = obj.attrs["EndGroup"]
-      except:
+      try:
+        self.allGroupEnds[name]  = obj.attrs["EndGroup"]
+      except KeyError:
```
Comment (joshua-cogliati-inl) [inline framework/h5py_interface_creator.py]: Okay, thank for fixing this.  If you want to back out the other commit ( git checkout c54daf02a114ac7a9975d6093e764bf8bf987aa5; git checkout -b alfoa/addedMemoryHandlingForBechlerLikeCluster ) I can approve it.

Comment (alfoa) [inline framework/h5py_interface_creator.py]: Done.

**Comment by joshua-cogliati-inl** [comment]
Merge Review Checklist
- [x] 1. Review all computer code. *Done*
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).  *No changes*
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://hpcgitlab.inl.gov/idaholab/raven/wikis/Developer_Information#python-code-standard) for details. *Yes*
- [ ] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass. *Waiting*
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True. *No functionality, just changing a if to an exception*
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync. *No changes*
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done. *It does*
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added? *No changes*

Once the automated tests pass, this is ready to be accepted.

**Comment by joshua-cogliati-inl** [comment]
The automated tests passed.

**Pull request merged**